service/controller/resource/ingress/resource.go

```diff
@@ -0,0 +1,140 @@
+package ingress
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/giantswarm/prometheus-meta-operator/service/controller/resource/generic"
+	"github.com/giantswarm/prometheus-meta-operator/service/key"
+)
+
+const (
+	Name = "ingress"
+)
+
+type Config struct {
+	K8sClient               k8sclient.Interface
+	Logger                  micrologger.Logger
+	BaseDomain              string
+	LetsencryptEnabled      bool
+	RestrictedAccessEnabled bool
+	WhitelistedSubnets      string
+}
+
+func New(config Config) (*generic.Resource, error) {
+	clientFunc := func(namespace string) generic.Interface {
+		c := config.K8sClient.K8sClient().ExtensionsV1beta1().Ingresses(namespace)
+		return wrappedClient{client: c}
+	}
+
+	c := generic.Config{
+		ClientFunc: clientFunc,
+		Logger:     config.Logger,
+		Name:       Name,
+		GetObjectMeta: func(v interface{}) (metav1.ObjectMeta, error) {
+			return getObjectMeta(v, config)
+		},
+		GetDesiredObject: func(v interface{}) (metav1.Object, error) {
+			return toIngress(v, config)
+		},
+		HasChangedFunc: hasChanged,
+	}
+	r, err := generic.New(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return r, nil
+}
+
+func getObjectMeta(v interface{}, config Config) (metav1.ObjectMeta, error) {
+	cluster, err := key.ToCluster(v)
+	if err != nil {
+		return metav1.ObjectMeta{}, microerror.Mask(err)
+	}
+
+	annotations := map[string]string{
+		"kubernetes.io/ingress.class":                "nginx",
+		"nginx.ingress.kubernetes.io/auth-signin":    "https://$host/oauth2/start?rd=$escaped_request_uri",
+		"nginx.ingress.kubernetes.io/auth-url":       "https://$host/oauth2/auth",
+		"nginx.ingress.kubernetes.io/rewrite-target": "/",
+		"kubernetes.io/tls-acme":                     fmt.Sprintf("%t", config.LetsencryptEnabled),
```

Comment:
Just out of curiosity, when is that false? I.e. when do we not want TLS enabled?

Reply:
Some installations do not support letsencrypt so the customers provide us their own certificates hence this become false

Reply:
Thanks, I didn't know about that.